scala/sort/insertion_sort.sc

```diff
@@ -0,0 +1,44 @@
+object InsertionSort {
+    private val notInOrder = (x1: Int, x2: Int, desc: Boolean) => {
```

Comment:
Sorry for the late review, it will be cleaner if we don't work with negations in method names 😅